apps/wallet/src/ui/app/helpers/sortAccounts.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { type AccountType, type SerializedUIAccount } from '_src/background/accounts/Account';
+import { isLedgerAccountSerializedUI } from '_src/background/accounts/LedgerAccount';
+import { isMnemonicSerializedUiAccount } from '_src/background/accounts/MnemonicAccount';
+import { isZkAccountSerializedUI } from '_src/background/accounts/zk/ZkAccount';
+
+const sortFns: Partial<
+	Record<AccountType, (a: SerializedUIAccount, b: SerializedUIAccount) => number>
+> = {
+	ledger: (a, b) =>
+		isLedgerAccountSerializedUI(a) && isLedgerAccountSerializedUI(b)
+			? a.derivationPath.localeCompare(b.derivationPath)
+			: 0,
+	'mnemonic-derived': (a, b) =>
+		isMnemonicSerializedUiAccount(a) && isMnemonicSerializedUiAccount(b)
+			? a.derivationPath.localeCompare(b.derivationPath)
+			: 0,
+};
+
+function getKey(account: SerializedUIAccount): string {
+	if (isMnemonicSerializedUiAccount(account)) return account.sourceID;
+	if (isZkAccountSerializedUI(account)) return account.provider;
+	return account.type;
+}
+
+export const defaultSortOrder: AccountType[] = [
+	'zk',
+	'mnemonic-derived',
+	'imported',
+	'ledger',
+	'qredo',
+];
+
+export function groupByType(accounts: SerializedUIAccount[]) {
+	return accounts.reduce(
+		(acc, account) => {
+			const byType = acc[account.type] || (acc[account.type] = {});
+			const key = getKey(account);
+
+			(byType[key] || (byType[key] = [])).push(account);
+
+			byType[key].sort(
+				sortFns[account.type] ??
+					((a: SerializedUIAccount, b: SerializedUIAccount) => a.address.localeCompare(b.address)),
```

Comment:
might be beneficial to add a creation time to the accounts so we can order by that instead of the address which a new one can go anywhere in the list. Just a thought, what do you think?

Reply:
yep - I also had that thought while implementing. I'll look into adding it!